Packet++/src/Sll2Layer.cpp

```diff
@@ -0,0 +1,203 @@
+	}
+
+	uint8_t macAddrAsArr[6];
+	macAddr.copyTo(macAddrAsArr);
+	return setLinkLayerAddr(macAddrAsArr, 6);
+}
+
+void Sll2Layer::parseNextLayer()
+{
+	if (m_DataLen <= sizeof(sll2_header))
+		return;
+
+	uint8_t* payload = m_Data + sizeof(sll2_header);
+	size_t payloadLen = m_DataLen - sizeof(sll2_header);
+
+	sll2_header* hdr = getSll2Header();
+	switch (be16toh(hdr->protocol_type))
+	{
+	case PCPP_ETHERTYPE_IP:
+		m_NextLayer = IPv4Layer::isDataValid(payload, payloadLen)
+			? static_cast<Layer*>(new IPv4Layer(payload, payloadLen, this, m_Packet))
+			: static_cast<Layer*>(new PayloadLayer(payload, payloadLen, this, m_Packet));
+		break;
+	case PCPP_ETHERTYPE_IPV6:
+		m_NextLayer = IPv6Layer::isDataValid(payload, payloadLen)
+			? static_cast<Layer*>(new IPv6Layer(payload, payloadLen, this, m_Packet))
+			: static_cast<Layer*>(new PayloadLayer(payload, payloadLen, this, m_Packet));
+		break;
+	case PCPP_ETHERTYPE_ARP:
+		m_NextLayer = new ArpLayer(payload, payloadLen, this, m_Packet);
+		break;
+	case PCPP_ETHERTYPE_VLAN:
+	case PCPP_ETHERTYPE_IEEE_802_1AD:
+		m_NextLayer = new VlanLayer(payload, payloadLen, this, m_Packet);
+		break;
+	case PCPP_ETHERTYPE_PPPOES:
+		m_NextLayer = PPPoESessionLayer::isDataValid(payload, payloadLen)
+			? static_cast<Layer*>(new PPPoESessionLayer(payload, payloadLen, this, m_Packet))
+			: static_cast<Layer*>(new PayloadLayer(payload, payloadLen, this, m_Packet));
+		break;
+	case PCPP_ETHERTYPE_PPPOED:
+		m_NextLayer = PPPoEDiscoveryLayer::isDataValid(payload, payloadLen)
+			? static_cast<Layer*>(new PPPoEDiscoveryLayer(payload, payloadLen, this, m_Packet))
+			: static_cast<Layer*>(new PayloadLayer(payload, payloadLen, this, m_Packet));
+		break;
+	case PCPP_ETHERTYPE_MPLS:
+		m_NextLayer = new MplsLayer(payload, payloadLen, this, m_Packet);
+		break;
+	default:
+		m_NextLayer = new PayloadLayer(payload, payloadLen, this, m_Packet);
+	}
+
+}
+
+void Sll2Layer::computeCalculateFields()
+{
+	if (m_NextLayer == nullptr)
+		return;
+
+	sll2_header* hdr = getSll2Header();
+	switch (m_NextLayer->getProtocol())
+	{
+		case IPv4:
+			hdr->protocol_type = htobe16(PCPP_ETHERTYPE_IP);
+			break;
+		case IPv6:
+			hdr->protocol_type = htobe16(PCPP_ETHERTYPE_IPV6);
+			break;
+		case ARP:
+			hdr->protocol_type = htobe16(PCPP_ETHERTYPE_ARP);
+			break;
+		case VLAN:
+			hdr->protocol_type = htobe16(PCPP_ETHERTYPE_VLAN);
+			break;
+		default:
+			return;
+	}
+}
+
+bool Sll2Layer::isDataValid(const uint8_t* data, size_t dataLen)
+{
+	return data && dataLen >= sizeof(sll2_header);
+}
+
+std::string Sll2Layer::toString() const
+{
+	return "Linux cooked header v2";
+}
+
+uint16_t sll2_header::getProtocolType() const
+{
+	return protocol_type;
+}
+
+void sll2_header::setProtocolType(uint16_t protocolType)
+{
+	protocol_type = protocolType;
+}
+
+uint16_t sll2_header::getReservedType() const
+{
+	return reserved_type;
+}
+
+void sll2_header::setReservedType(uint16_t reservedType)
+{
+	reserved_type = reservedType;
+}
```

Comment:
When creating getters (or setters) with a value that is larger than `uint8_t` it'd be nicer to do the conversion from network->host byte order (or host->network) with the method. For example:

```cpp
uint16_t sll2_header::getReservedType() const
{
	return betoh16(reserved_type);
}

void sll2_header::setReservedType(uint16_t reservedType)
{
	reserved_type = htobe16(reservedType);
}
```

Please do the same to all getters and setter that are larger than `uint8_t`

Reply:
done